ote/src/cljs/ote/views/transport_operator.cljs

```diff
@@ -228,106 +230,93 @@
+        {:type :result-msg-warning
+         :element-id "label-public-contact-missing"
+         :name :public-contact-missing
+         :content (tr [:common-texts :required-one-of-many])}
+        {:type :text-label
+         :name :warning-public-contact-missing-placeholder
+         :label ""
+         :h-style :h4})
+
+      {:type :string
+       :element-id "input-operator-telephone"
        :name ::t-operator/phone
        :label (tr [:organization-page :field-phone-telephone] )
-       :type :string
        :disabled? (get-in state [:ytj-flags :use-ytj-phone?] false)
+       :required? (required-public-contact-missing? operator)
        :style style-fields/form-field
        :regex ui-validation/phone-number-regex}
 
-      {:element-id "input-operator-mobilePhone"
+      {:type :string
+       :element-id "input-operator-mobilePhone"
        :name ::t-operator/gsm
        :label (tr [:organization-page :field-phone-mobile] )
-       :type :string :disabled? (get-in state [:ytj-flags :use-ytj-gsm?] false)
+       :disabled? (get-in state [:ytj-flags :use-ytj-gsm?] false)
+       :required? (required-public-contact-missing? operator)
        :style style-fields/form-field
        :regex ui-validation/phone-number-regex}
 
-      {:element-id "input-operator-email"
+      {:type :string
+       :element-id "input-operator-email"
        :name ::t-operator/email
-       :type :string
        :disabled? (get-in state [:ytj-flags :use-ytj-email?] false)
+       :required? (required-public-contact-missing? operator)
        :style style-fields/form-field}
 
-      {:element-id "input-operator-web"
+      {:type :string
+       :element-id "input-operator-web"
        :name ::t-operator/homepage
-       :type :string
        :disabled? (get-in state [:ytj-flags :use-ytj-homepage?] false)
-       :style style-fields/form-field})))
+       :required? (required-public-contact-missing? operator)
+       :style style-fields/form-field}
+
+      ;; Contact details for authorities
+
+      {:type :text-label
+       :name :heading-contact-details-authorities
+       :label (tr [:organization-page :contact-details-authorities])
+       :h-style :h2}
+
+      {:type :info-toggle
+       :name :help-operator-contact-details-auth
+       :label (tr [:common-texts :instructions])
+       :body [:p (tr [:organization-page :help-contact-details-auth])]
+       :default-state true}
+
+      {:type :string
+       :element-id "input-operator-telephone-auth"
+       :name ::t-operator/authority-phone
+       :required? true
+       :label (tr [:organization-page :field-phone-telephone])
+       :style style-fields/form-field
+       :regex ui-validation/phone-number-regex}
+
+      {:type :string
+       :element-id "input-operator-email-auth"
+       :name ::t-operator/authority-email
+       :required? true
+       :label ::t-operator/email
+       :style style-fields/form-field
+       ;:validate [[:email]];;TODO implement email regex
```

Comment:
This seems to work quite well:
(defn validate-email
  [email]
  (let [pattern #"[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?"]
    (and (string? email) (re-matches pattern email))))

Reply:
True. Did not decide to implement it here because this will use the implementation coming from transport service changes which are ongoing parallel. That will be taken into use when it's merged.